src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/FullBlockSectionProcessor.scala

```diff
@@ -87,6 +108,30 @@ trait FullBlockSectionProcessor extends BlockSectionProcessor with FullBlockProc
         }
         .result
     }
+
+    /**
+      * Validation specific to concrete type of block payload
+      */
+    private def modifierSpecificValidation(m: BlockSection, header: Header): ValidationState[Unit] = {
+      m match {
+        case e: Extension =>
+          // todo checks that all required mandatory fields are set
+          // todo checks number of mandatory fields
+          failFast
+            .validate(e.mandatoryFields.forall(_._1.lengthCompare(Extension.MandatoryFieldKeySize) == 0)) {
+              fatal(s"Extension ${m.encodedId} mandatory field key length is not ${Extension.MandatoryFieldKeySize}")
+            }
+            .validate(e.optionalFields.forall(_._1.lengthCompare(Extension.OptionalFieldKeySize) == 0)) {
+              fatal(s"Extension ${m.encodedId} optional field key length is not ${Extension.OptionalFieldKeySize}")
+            }
+            .validate(e.optionalFields.lengthCompare(Extension.MaxOptionalFields) <= 0) {
+              fatal(s"Extension ${m.encodedId} have too many optional fields")
+            }
```

Comment:
No value size check for optional (and, btw, mandatory, fields), It may be reasonable also to fail fast on max size limit being reached.